src/plugin.js

```diff
@@ -255,17 +255,44 @@ const initPlugin = function(player, options) {
 
     content.className = 'vjs-errors-dialog';
     content.id = 'vjs-errors-dialog';
+
+    const errorCode = `<div class="vjs-errors-code"><b>${this.localize('Error Code')}:</b> ${(error.type || error.code)}</div>`;
+    const isTimeoutError = error.code === -2;
+
     dialogContent =
      `<div class="vjs-errors-content-container">
-      <h2 class="vjs-errors-headline">${this.localize(error.headline)}</h2>
-        <div class="vjs-errors-code"><b>${this.localize('Error Code')}:</b> ${(error.type || error.code)}</div>
-        ${details}
+        <h2 class="vjs-errors-headline">${this.localize(error.headline)}</h2>
+        ${isTimeoutError ? '' : errorCode}
+        ${isTimeoutError ? '' : details}
       </div>`;
 
     const closeable = display.closeable(!('dismiss' in error) || error.dismiss);
 
-    // We should get a close button
-    if (closeable) {
+    if (isTimeoutError) {
+      dialogContent +=
+       `<div class="vjs-errors-timeout-button-container">
+         <button>${this.localize('Reload Video')}</button>
+         <button>${this.localize('Dismiss')}</button>
+       </div>`;
+      content.innerHTML = dialogContent;
+      display.fillWith(content);
+
+      display.getChild('closeButton').hide();
+
+      const reloadButton = display.el().querySelector('.vjs-errors-timeout-button-container > button:first-child');
+      const dismissButton = display.el().querySelector('.vjs-errors-timeout-button-container > button:last-child');
+
+      player.on(reloadButton, 'click', function() {
+        const source = player.currentSource();
+
+        player.reset();
+        player.src(source);
+      });
+      player.on(dismissButton, 'click', function() {
+        display.close();
+      });
```

Comment:
Unfortunately, these listeners create a small memory leak. Even when the display is closed and the elements are in the DOM, the internal Video.js listeners do not go away.

There's not a very easy way to find this, but I set a breakpoint in Video.js and exposed the `DomData` object. Then I tracked the number of listeners counted on this object as I created and dismissed the timeout modal multiple times. Each time the number of listeners increased by two. 

I'd suggest adding the following:
```js
display.one('modalclose', () => {
  player.off(reloadButton);
  player.off(dismissButton);
});
```
That should clear these two listeners when these DOM elements disappear when the display is closed.